src/main/java/io/openmarket/marketplace/MarketPlaceServiceHandler.java

```diff
@@ -0,0 +1,179 @@
+package io.openmarket.marketplace;
+
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.ConditionalCheckFailedException;
+import com.amazonaws.services.dynamodbv2.model.UpdateItemRequest;
+import com.google.common.collect.ImmutableMap;
+import io.openmarket.marketplace.dao.ItemDao;
+import io.openmarket.marketplace.grpc.MarketPlaceProto.*;
+import io.openmarket.marketplace.model.Item;
+import io.openmarket.transaction.grpc.TransactionProto;
+import io.openmarket.transaction.model.TransactionStatus;
+import io.openmarket.transaction.service.TransactionServiceHandler;
+import lombok.extern.log4j.Log4j2;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.util.*;
+
+import static io.openmarket.config.MerchandiseConfig.*;
+
+@Log4j2
+public class MarketPlaceServiceHandler {
+    private final ItemDao itemDao;
+    private final TransactionServiceHandler transactionServiceHandler;
+
+    //the checkout is aborted after checking transaction status for 20 second
+    private final int CHECKOUT_TIMEOUT = 20;
+    private final int CHECKOUT_QUERY_DELAY = 2000;
+    @Inject
+    public MarketPlaceServiceHandler(@Nonnull final ItemDao itemDao,
+                                     @Nonnull final TransactionServiceHandler transactionServiceHandler) {
+        this.itemDao = itemDao;
+        this.transactionServiceHandler = transactionServiceHandler;
+    }
+
+
+    public GetOrgItemsResult getListingByOrgId(GetOrgItemsRequest request) {
+        List<String> itemIds = this.itemDao.getItemIdsByOrg(request.getOrgId());
+        List<Item> selling = this.itemDao.batchLoad(itemIds);
+        List<ItemGrpc> result = new ArrayList<>();
+
+        for(Item item : selling) {
+            result.add(
+                    ItemGrpc.newBuilder()
+                    .setItemName(item.getItemName()).setItemStock(item.getStock())
+                    .setBelongTo(item.getBelongTo())
+                    .setItemPrice(item.getItemPrice())
+                    .setItemDescription(item.getItemDescription())
+                    .setItemId(item.getItemID())
+                    .setItemImageLink(item.getItemImageLink())
+                    .build()
+            );
+        }
+
```

Comment:
Future improvement: Consider using grpc stream for returning a potentially long list of items.